src/test/java/integration/TourOfferServiceTest.java

```diff
@@ -0,0 +1,120 @@
+package integration;
+
+import com.epam.exception.NotFoundException;
+import com.epam.model.TourOffer;
+import com.epam.repository.TourOfferDAO;
+import com.epam.service.TourOfferService;
+import java.time.LocalDate;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class TourOfferServiceTest {
+
+    private TourOfferService tourOfferService;
+
+    @Mock
+    private JdbcTemplate jdbcTemplate;
+    @Mock
+    private TourOffer tourOffer;
+
+    private List<TourOffer> tourOffers;
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+
+        tourOfferService = new TourOfferService(new TourOfferDAO(jdbcTemplate));
+        tourOffers = new ArrayList<>();
+        tourOffers.add(tourOffer);
+    }
+
+    @Test
+    public void getAll() {
+        Mockito.when(jdbcTemplate.query(Mockito.anyString(), Mockito.any(RowMapper.class))).thenReturn(tourOffers);
+        List<TourOffer> actualTours = tourOfferService.getTours();
+        Assert.assertEquals(tourOffer, actualTours.get(0));
+    }
+
+    @Test
+    public void getAllEmptyList() {
+        Mockito.when(jdbcTemplate.query(Mockito.anyString(), Mockito.any(RowMapper.class))).thenReturn(Collections.emptyList());
+        List<TourOffer> actualTours = tourOfferService.getTours();
+        Assert.assertEquals(0, actualTours.size());
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void deleteTourNullId() {
+        tourOfferService.deleteTour(null);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void deleteTourZeroId() {
+        tourOfferService.deleteTour(0);
+    }
+
+    @Test
+    public void deleteTour() {
+        int tourOfferId = 10;
+        Mockito.when(jdbcTemplate.update(Mockito.anyString(), (Integer) Mockito.any())).thenReturn(1);
+
+        int actualReturnCode = tourOfferService.deleteTour(tourOfferId);
+        Assert.assertEquals(1, actualReturnCode);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void addTourNull() {
+        tourOfferService.addTour(null);
+    }
+
+//    @Test(expected = IllegalArgumentException.class)
+//    public void addTourOfferWrongHotelId() {
```

Comment:
Oh, no. please. Let's do not commit commented code

Reply:
These tests should be uncommented after fix bugs

Reply:
I add tasks on the board and write tests now to test them in future